test/boost/sstable_datafile_test.cc

```diff
@@ -1036,6 +1036,76 @@ static flat_mutation_reader make_normalizing_sstable_reader(shared_sstable sst,
     return make_normalizing_sstable_reader(sst, s, query::full_partition_range);
 }
 
+SEASTAR_TEST_CASE(autocompaction_control_test) {
+    return test_env::do_with_async([] (test_env& env) {
+        storage_service_for_tests ssft;
+        BOOST_REQUIRE(smp::count == 1);
+
+        auto cm = make_lw_shared<compaction_manager>();
+        cm->start();
+
+        schema_builder builder(make_lw_shared(schema({}, some_keyspace, some_column_family,
+            {{"p1", utf8_type}}, {{"c1", utf8_type}}, {{"r1", utf8_type}}, {}, utf8_type)));
+        builder.set_min_compaction_threshold(2);
+        auto s = builder.build(schema_builder::compact_storage::yes);
+
+        auto tmp = tmpdir();
+        column_family::config cfg = column_family_test_config();
+        cfg.datadir = tmp.path().string();
+        cfg.enable_commitlog = false;
+        cfg.enable_incremental_backups = false;
+        auto cl_stats = make_lw_shared<cell_locker_stats>();
+        auto tracker = make_lw_shared<cache_tracker>();
+        auto cf = make_lw_shared<column_family>(s, cfg, column_family::no_commitlog(), *cm, *cl_stats, *tracker);
+        cf->start();
+        cf->mark_ready_for_writes();
+        cf->set_compaction_strategy(sstables::compaction_strategy_type::size_tiered);
+
+        // create and add sstables
+        auto sst_gen = [&env, s, &tmp, gen = make_lw_shared<unsigned>(1)] () mutable {
+            return env.make_sstable(s, tmp.path().string(), (*gen)++, la, big);
+        };
+
+        auto make_insert = [&] (partition_key key) {
+            mutation m(s, key);
+            m.set_clustered_cell(clustering_key::from_exploded(*s, {to_bytes("c1")}), bytes("r1"), data_value(int32_t(1)), 1);
+            return m;
+        };
+
+        auto alpha = partition_key::from_exploded(*s, {to_bytes("alpha")});
+
+        column_family_test(cf).add_sstable(make_sstable_containing(sst_gen, {make_insert(alpha)}));
+        column_family_test(cf).add_sstable(make_sstable_containing(sst_gen, {make_insert(alpha)}));
```

Comment:
By adding two SSTables, there is a risk a compaction will happen. In particular, you set the threshold to two, so I have no idea why it doesn't happen. Maybe it does happen and that's another point where the test can get flaky.

It's best to disable the compaction first, then add the SSTables.

Reply:
:+1: